Revise obs space localization for LETKF

rtodling opened the pull request:
## Description

The original lenghscales were generating a completely noisy, unphysical, increment. 

I don't claim scientific correctness ... just reasonableness. Here is the T-500 increment original and after the revised scales:

<img width="659" alt="letkf_loc500" src="https://github.com/user-attachments/assets/23ef3fe4-c72d-46b8-8faf-c723455f3ba5">

Note these results use 32 members, at c90 resolution. I will suggest we change to more than 2 members for testing purposes in swell - in another PR - I don't think 2 members makes any sense.

## Dependencies

None

## Impact

None

rtodling committed:
revise obs space localization for LETKF

Dooruk reviewed (dismissed):
Looks good to me, I will wait for @metdyn's approval

rtodling commented:
@Dooruk  let me make one more change to this ...

rtodling committed:
use all ob system

rtodling commented:
I add all obs systems to:

test/suite_tests/localensembleda-tier1.yaml

but I cannot understand why the job fails when trying to add the obs localization:

/gpfsm/dnb05/projects/p139/rtodling/cylc-run/swell-getkf-all-suite/run1/log/job/20211212T0000Z/RunJediLocalEnsembleDaExecutable-geos_atmosphere/01/job.err

rtodling commented:
It seems the localization directory is not being installed ...

rtodling committed:
this was missing

rtodling commented:
I think I got the localization directory installed now ... changed the setup.py

Dooruk commented:
~You need to edit https://github.com/GEOS-ESM/swell/blob/develop/setup.py~
Beat me to it

rtodling commented:
32 members seem to be too much for the default PE choice ... trying w/ 10 members

rtodling committed:
works but ...

rtodling commented:
@Dooruk @metdyn This works - not ideal cause only a few instruments and very few members - but it runs ... the problem being that it we need to add something in the yaml handling logic for the obs that tells the code to bypass the writing of HofXs - for now: that is, for example, for aircraft, we need to bypass:

      obsdataout:
        engine:
          obsfile: /discover/nobackup/rtodling/SwellExperiments/SLES15/swell-getkf-all/run/20211212T000000Z/geos_atmosphere/swell-getkf-all.aircraft.20211211T210000Z.nc4

this means that we'll need to have if statements around each observation file projecting this for "if not ensemble" then write files out ... 


also, I tried changing the defaults:

  save posterior ensemble: true
  save posterior ensemble increments: true
  save posterior mean: false
  save posterior mean increment: true


to simply:

  save posterior ensemble: false
  save posterior ensemble increments: false
  save posterior mean: false
  save posterior mean increment: true 

but I had troubles w/ Swell not liking what I did ... perhaps @metdyn you can look at these issues ... and add to what here in this branch?

metdyn commented:
Thank you and I will look at this.
That the yaml control did not work is really inconvenient and I will check
that.


On Thu, Dec 5, 2024 at 2:52 PM rtodling ***@***.***> wrote:

> @Dooruk <https://github.com/Dooruk> @metdyn <https://github.com/metdyn>
> This works - not ideal cause only a few instruments and very few members -
> but it runs ... the problem being that it we need to add something in the
> yaml handling logic for the obs that tells the code to bypass the writing
> of HofXs - for now: that is, for example, for aircraft, we need to bypass:
>
>   obsdataout:
>     engine:
>       obsfile: /discover/nobackup/rtodling/SwellExperiments/SLES15/swell-getkf-all/run/20211212T000000Z/geos_atmosphere/swell-getkf-all.aircraft.20211211T210000Z.nc4
>
> this means that we'll need to have if statements around each observation
> file projecting this for "if not ensemble" then write files out ...
>
> also, I tried changing the defaults:
>
> save posterior ensemble: true
> save posterior ensemble increments: true
> save posterior mean: false
> save posterior mean increment: true
>
> to simply:
>
> save posterior ensemble: false
> save posterior ensemble increments: false
> save posterior mean: false
> save posterior mean increment: true
>
> but I had troubles w/ Swell not liking what I did ... perhaps @metdyn
> <https://github.com/metdyn> you can look at these issues ... and add to
> what here in this branch?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GEOS-ESM/swell/pull/477#issuecomment-2521266871>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIVBWOASJTJ7UFVBIW27BW32ECVGNAVCNFSM6AAAAABTBGYFO2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMRRGI3DMOBXGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

metdyn committed:
Help with PR477 (#479)

* ...

* Add to localensembleda-tier1.yaml
    local_ensemble_save_posterior_ensemble: false
    local_ensemble_save_posterior_ensemble_increments: false
    local_ensemble_save_posterior_mean: false
    local_ensemble_save_posterior_mean_increment: true
output jedi_localensembleda_config.yaml shows right.

* In run_jedi_local_ensemble_da_executable.py, hard code if bypass_writing_hofxs:  del [obs space][obsdataout]

* update default for save_posterior_ensemble

* add back lienar_observer: true

metdyn commented on setup.py:
```diff
@@ -45,6 +45,7 @@
                'configuration/*/*/*',
                'configuration/*/*/*/*',
                'configuration/*/*/*/*/*',
+               'configuration/*/*/*/*/*/*',
```
@rtodling @Dooruk ,
If you donot mind, I will delete this line.
Without it, I already get to the step jedi_localensembleda_log.log in cylc. 
``` 

LinearModel<MODEL>::forecastTL: Starting
----------------------------------------------------------------------------------------------------
Increment print | number of fields = 8 | cube sphere face size: C90
```

Dooruk replied:
@metdyn I believe @rtodling included that for `Localizations` folder to work. If you are %100 sure that it's not needed I don't care.

metdyn replied:
Thank you, @rtodling, for the great work!   We should merge this long due PR, @Dooruk! (No, this line does not bother me either)

metdyn reviewed (dismissed):
Thanks a lot for the great work!

metdyn committed:
Merge branch 'develop' into feature/rtodling/revise_letkf_oblocalization
